esmf_regrid/esmf_regridder.py

```diff
@@ -294,6 +296,13 @@ def regrid(self, src_array, norm_type="fracarea", mdtol=1):
             A numpy array whose shape is compatible with self.tgt.
 
         """
+        array_shape = src_array.shape
+        main_shape = array_shape[-self.src.dims :]
+        if main_shape != self.src.shape:
```

Comment:
It's worth considering the case where one (or more) of the grid dimensions is a scalar. We may end up comparing (n) with (1, n) and raising an error where we would otherwise want to be lenient.

Reply:
Having discussed this, we think this is needed but will be done in a separate PR as it doesn't impact the answers here.